code/modules/surgery/organs/augments_arms.dm

```diff
@@ -411,3 +411,128 @@
+		return FALSE
+	..()
+
+/obj/item/organ/internal/cyberimp/arm/v1_arm/Retract()
+	if(disabled)
+		to_chat(owner, "<span class='warning'>Your arm fails to retract!</span>")
+		return FALSE
+	..()
+
+/obj/item/shield/v1_arm
+	name = "vortex feedback arm" //format is they are holding x
+	desc = "A modification to a users arm, allowing them to use a vortex core energy feedback, to parry, reflect, and even empower projectile attack. Rumors that it runs on the users blood are unconfirmed"
+	icon_state = "v1_arm"
+	item_state = "v1_arm"
+	sprite_sheets_inhand = list("Drask" = 'icons/mob/clothing/species/drask/held.dmi', "Vox" = 'icons/mob/clothing/species/vox/held.dmi')
+	force = 20 //bonk, not sharp
+	attack_verb = list("slamed", "punched", "parried", "judged", "styled on", "disrespected", "interupted", "gored")
+	hitsound = 'sound/effects/bang.ogg'
+	light_power = 3
+	light_range = 0
+	light_color = "#9933ff"
+	hit_reaction_chance = -1
+	/// The damage the reflected projectile will be increased by
+	var/reflect_damage_boost = 10
+	/// The cap of the reflected damage. Damage will not be increased above 50, however it will not be reduced to 50 either.
+	var/reflect_damage_cap = 50
+	var/disabled = FALSE
+	var/force_when_disabled = 5 //still basically a metal pipe, just hard to move
+
+
+/obj/item/shield/v1_arm/emp_act(severity)
+	if(disabled)
+		return
+	to_chat(loc, "<span class='warning'>Your arm seises up!</span>")
+	disabled = TRUE
+	force = force_when_disabled
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 10 SECONDS)
+
+/obj/item/shield/v1_arm/proc/reboot()
+	disabled = FALSE
+	force = initial(force)
+
+/obj/item/shield/v1_arm/add_parry_component()
+	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.35, _parryable_attack_types = ALL_ATTACK_TYPES, _parry_cooldown = (1 / 3) SECONDS, _no_parry_sound = TRUE) // 0.3333 seconds of cooldown for 75% uptime, countered by ions and plasma pistols
+
+/obj/item/shield/v1_arm/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(disabled)
+		return FALSE
+	// Hit by a melee weapon or blocked a projectile
+	. = ..()
+	if(.) // they did parry the attack
+		// Hit by a projectile
+		if(. > 1) // a perfect parry
+			set_light(3)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/atom, set_light), 0), 0.25 SECONDS)
+
+			if(istype(hitby, /obj/item/projectile))
+				var/obj/item/projectile/P = hitby
+				if(P.shield_buster || istype(P, /obj/item/projectile/ion)) //EMP's and unpariable attacks, after all.
+					return FALSE
+				if(P.reflectability == REFLECTABILITY_NEVER) //only 1 magic spell does this, but hey, needed
+					owner.visible_message("<span class='danger'>[owner] blocks [attack_text] with [src]!</span>")
+					playsound(src, 'sound/weapons/effects/ric3.ogg', 100, TRUE)
+					return TRUE
+
+				P.damage = max(min(reflect_damage_cap, P.damage += 10), P.damage)
```

Comment:
```suggestion
				P.damage = clamp((P.damage + 10), P.damage, reflect_damage_cap)
```
I think this is what you meant to do here? If you use `max(min(` you can use the `clamp()` proc instead to do the same thing cleaner